src/Network/GRPC/MQTT/Message/Packet.hs

```diff
@@ -158,55 +164,110 @@ makePacketReader queue = do
 
 -- Packet - Packet Senders -----------------------------------------------------
 
+newtype PacketSizeError = PacketSizeError
+  {getPacketSizeError :: Integer}
+  deriving (Data, Eq, Generic, Ord, Show, Typeable)
+
+-- | @since 1.0.0
+instance Exception PacketSizeError where
+  displayException (PacketSizeError n) =
+    Text.printf
+      "%s: maximum packet payload size must be between %d and %d bytes."
+      (show (PacketSizeError n) :: String)
+      minPacketSize
+      maxPacketSize
+  {-# INLINE displayException #-}
+
+-- | The predicate 'isValidPacketSize' tests if the 'Word32' can be used as the
+-- maximum packet payload size in 'makePacketSender'. This is any 'Word32' value
+-- that is:
+--
+-- * Greater than or equal to 'minPacketSize'.
+--
+-- * Strictly less than 'maxPacketSize'.
+--
+-- @since 1.0.0
+isValidPacketSize :: Word32 -> Bool
+isValidPacketSize x = minPacketSize <= x && x < maxPacketSize
+
+-- | The minimum packet payload size, 16 bytes.
+--
+-- @since 1.0.0
+minPacketSize :: Word32
+minPacketSize = 16
```

Comment:
This is the length of the serialization overhead introduced by `position` and `npackets` round upwards towards the nearest power of 2 (i.e. 12 -> 16).

Reply:
I see. I still think it should be defined in terms of the size of an empty `Packet` so that someone in the future, modifying the `Packet` structure, can't forget to update the `minPacketSize`.